src/lib.rs

```diff
@@ -601,3 +597,149 @@ impl<'lib> IRCompiler<'lib> {
         }
     }
 }
+
+// Moved from main.rs to lib.rs
+fn create_static_sampler(
+    min_mag_mip_mode: IRFilter,
+    address_mode: IRTextureAddressMode,
+    index: u32,
+    anisotropy: Option<u32>,
+) -> IRStaticSamplerDescriptor {
+    let max_anisotropy = anisotropy.unwrap_or(1);
+
+    IRStaticSamplerDescriptor {
+        filter: min_mag_mip_mode,
+        address_u: address_mode,
+        address_v: address_mode,
+        address_w: address_mode,
+        mip_lod_bias: 0.0,
+        max_anisotropy: max_anisotropy,
+        comparison_func: IRComparisonFunction::IRComparisonFunctionNever,
+        min_lod: 0.0,
+        max_lod: 100000.0,
+        shader_register: index,
+        register_space: 0,
+        shader_visibility: IRShaderVisibility::IRShaderVisibilityAll,
+        border_color: IRStaticBorderColor::IRStaticBorderColorTransparentBlack,
+    }
+}
+
+/// Takes a DXIL binary, cross-compiles it to metal and returns a metallib binary
+pub fn compile_dxil_to_metallib(
+    path_compiler_lib: &Path,
+    dxil_binary: &Vec<u8>,
```

Comment:
Clippy will probably complain that this should be a slice, instead of a borrow of a forcibly-owned `Vec`.